spam/spam_tutorial.py

```diff
@@ -140,234 +145,406 @@
+# * **Aim to have overlaps.** We'll train a _label model_ to denoise and combine the outputs of our LFs. The label model learns from overlaps and conflicts between LFs, so it's helpful if multiple LFs cover certain areas of the data set.
 #
-# Once multiple LFs have been created, users can look at data points that have received no labels so far (or many conflicting labels) to get ideas for new LFs to write.
-# Furthermore, if there are some classes that have votes from very few LFs, users might iterate by writing additional LFs to cover those parts of the dataset.
-# **Note, however, that it is not necessary for LFs to assign labels to every data point;** in fact, most of the time your LFs will not have perfect dataset-wide coverage.
-# We rely on the fact that the classifier that trains on labels from Snorkel has the power to _generalize_ and can therefore learn a good representation of the data even if the each data point in the training set does not receive a label from any LFs.
-# In addition, note that **it is ok to have overlaps and conflicts between labeling functions**. For example, two good labeling functions can conflict when one of them is written to fix the errors of the other. We later use a _label model_ that learns how to combine the outputs of all (potentially conflicting) labeling functions to estimate label probabilities.
+# Of course, rules are meant to be broken.
+# For example, we can sometimes increase performance by adding LFs that label only if certain others don't.
+# And in some advanced problems, a 95% accurancy and 10% coverage LF may in fact be preferable to a 90% accurancy and 30% coverage one.
+# Use the principles above to guide your process, but don't treat them as hard-and-fast rules.
+#
+# Our goal for LF development is to create a high quality set of training labels for our unlabeled data set,
+# not to label everything or directly create a model for inference using the LFs.
+# The training labels are used to train a separate discriminative model (in this case, one which just uses the comment text) in order to generalize to new, unseen examples.
+# Using this model, we can make predictions for data points that our LFs don't cover.
+#
+# We'll walk through the development of two LFs using basic analysis tools in Snorkel, then provide a full set of LFs that we developed for this tutorial.
 
 # %% [markdown]
-# ### a) Look at examples for ideas
+# ### a) Initial ideation: look at training examples
 
 # %% [markdown]
-# We begin the process of writing LFs by looking at some examples in the dev set.
+# Following recommended practice above, we'll start by looking at the `train` set to generate some ideas for LFs.
```

Comment:
I think it's easier to look at the dev set- that was the original intended purpose (hence 'dev'; in fact idea was dev set was created while looking at data during development), and it's clearer for tutorial purposes bc then we see the labels

Reply:
And note that by looking at a new subset of the training set, you're implicitly creating a new labeled dev set; and this is making things too complicated